apps/hosts/views.py

```diff
@@ -197,13 +195,24 @@ def challenge_host_detail(request, challenge_host_team_pk, pk):
             response_data = serializer.data
             return Response(response_data, status=status.HTTP_200_OK)
         else:
-            return Response(
-                serializer.errors, status=status.HTTP_400_BAD_REQUEST
-            )
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
     elif request.method == "DELETE":
-        challenge_host.delete()
-        return Response(status=status.HTTP_204_NO_CONTENT)
+        if challenge_host_team.created_by == request.user:
+
+            if (challenge_host.user == request.user):  # when the user tries to remove himself
+                response_data = {
+                    "error": "You are not allowed to remove yourself since you are the team admin. Please delete the team if you want to do so!"
+                }  # noqa: ignore=E501
```

Comment:
Why are we using `#noqa:.. ` here?

Reply:
I took the reference of [this](https://github.com/Cloud-CV/EvalAI/blob/master/apps/participants/views.py#L281) API for writing this one and [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/participants/views.py#L304) as you can see the API was written probably by assuming that the check on whether the member to be deleted is admin or not is being done in the frontend only and so this condition will never be required in that case. But now as we are using this condition only to perform this check so will remove this comment from both the APIs.